### PR TITLE
Components: ImageZoom

### PR DESCRIPTION
**Pull request by peterhass**
https://jira.nebenan.de/browse/CORE-6199

Status:
- ✅  PM approved
- ✅. QA approved

### CODE REVIEW

File: packages/components/src/image_zoom/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom/hooks.jsx:70]: resize observer doesn't work in some mobile safari versions - add polyfill in apps?

File: packages/components/src/image_zoom/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom/hooks.jsx:129]: Open for suggestions how to further refactor this. It's important to use a reducer here because many state variables depend on one another

File: packages/components/src/image_zoom/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom/hooks.jsx:214]: just rename it to `zoom`

File: packages/components/src/image_zoom/index.jsx
Comment (peterhass) [inline packages/components/src/image_zoom/index.jsx:30]: time to extract imo

**Comment by peterhass** [comment]
Will follow up with tests after the first review round

File: packages/components/src/image_zoom/index.jsx
Comment (peterhass) [inline packages/components/src/image_zoom/index.jsx:22]: Why I had to add `maxHeight`: We previously talked about always wanting to show images in a 16:9 ratio but that's not really true. If you upload a picture, you want to show it in its original aspect ratio. Everything else wouldn't really make sense since it would give you quite a small 'view window' to see the image content. If we go with dynamic aspect ratio, we have to be able to limit the max height of the image - otherwise we prevent mobile users to properly use the gallery (they can't scroll down as easy as desktop users).

Open for suggestions for the implementation

File: packages/components/src/image_zoom_slider/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom_slider/hooks.jsx:3]: this needs to go somewhere else - nebenan-forms for example

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom_view/hooks.jsx:113]: listening for window resize may be enough

File: packages/components/src/image_zoom_provider/utils.jsx
Comment (suprMax) [inline packages/components/src/image_zoom_provider/utils.jsx:1]: lodash.clamp

File: packages/components/src/image_zoom_slider/index.jsx
Comment (suprMax) [inline packages/components/src/image_zoom_slider/index.jsx:27]: add a note to refactor Slider later: 
https://favro.com/organization/aee06c3fe6503c804f507dd9/22c630cc02b09f2d6f98b8b3?card=Goo-29771

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom_view/hooks.jsx:122]: TODO: find out why we use js to calculate this and add an comment

File: packages/components/src/image_zoom_view/index.jsx
Comment (peterhass) [inline packages/components/src/image_zoom_view/index.jsx:42]: check why nativeEvent is needed

File: packages/components/src/image_zoom_view/index.jsx
Comment (peterhass) [inline packages/components/src/image_zoom_view/index.jsx:72]: take 30min: dragging should happen too while pinch to zoom (check ios photos app)

File: packages/components/src/image_zoom_view/index.module.scss
Comment (suprMax) [inline packages/components/src/image_zoom_view/index.module.scss:5]: ?

File: packages/components/src/image_zoom_view/utils.jsx
Comment (suprMax) [inline packages/components/src/image_zoom_view/utils.jsx:1]: duplicated , clamp

File: packages/components/src/image_zoom_slider/index.jsx
Comment (suprMax) [inline packages/components/src/image_zoom_slider/index.jsx:33]: higher resolution?

**Comment by suprMax** [comment]
+ zoom out bug

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (peterhass) [inline packages/components/src/image_zoom_view/hooks.jsx:115]: maybe use `useLayoutEffect `with an guard? then we don't need the on-mount effect above

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_view/hooks.jsx:140]: see here https://github.com/goodhood-eu/goodhood/pull/69/files#diff-4c7fd048bb7e22351274bfacd37529bba804bce013b100180eaebae33074acd5R155

**Review by suprMax — COMMENTED** [review]
1. Check this error:
Screen Shot 2021-01-14 at 1 35 31 PM
2. Adjust step in double tap so it spreads 0-1 evenly

File: packages/components/src/image_zoom_provider/hooks.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_provider/hooks.jsx:2]: yes

File: packages/components/src/image_zoom_view/index.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_view/index.jsx:73]: touch move?

`pinchZoom` now handles offset moving too. No way `drag` could do that.

I really don't know how we should change this and why:

`touchStart`:
- 1 finger: drag + double tap
- 2 fingers: drag (to be ready when user removes one finger) + pinchZoom

`touchMove`:
- 1 finger: drag
- 2 fingers: pinchZoom

File: packages/components/src/image_zoom_view/index.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_view/index.jsx:87]: touchend gets called every time a touch is removed

- touchstart with 2 fingers 
- you move around -> touchmove
- you let go of one finger -> touchend (e.touches.length === 1)

Will add a comment to make it more clear.

File: packages/components/src/image_zoom_view/index.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_view/index.jsx:131]: we configured our linter to force `alt` attributes on `img` tags -> therefore we should go with that rule here too

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_view/hooks.jsx:13]: yes

File: packages/components/src/image_zoom_slider/hooks.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_slider/hooks.jsx:4]: still needs your review: https://github.com/goodhood-eu/nebenan-form/pull/47

**Comment by peterhass** [comment]
Gave the ticket to QA, let's wait for their result before we start the final review.

File: packages/components/src/image_zoom_view/index.jsx
Comment (suprMax) [reply packages/components/src/image_zoom_view/index.jsx:77]: ideally two finger drag should also work,

Comment (peterhass) [reply packages/components/src/image_zoom_view/index.jsx:77]: it's integrated in pichZoom - you cannot do two finger drag without pinch zoom

**Comment by peterhass** [comment]
QA approved. Please re-review @suprMax

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (suprMax) [inline packages/components/src/image_zoom_view/hooks.jsx:110]: probably debounce suits better?

**Review by suprMax — APPROVED** [review]
LGTM, see small comments

File: packages/components/src/image_zoom_view/hooks.jsx
Comment (peterhass) [reply packages/components/src/image_zoom_view/hooks.jsx:110]: I used debounce at first but actually throttle feels better. If you are slowly changing window size, debounce would wait for the resizing to stop.

**Pull request merged**
